src/containers/LoginScreen/index.js

```diff
@@ -2,38 +2,44 @@ import React, { memo, useCallback } from 'react';
 import { useDispatch } from 'react-redux';
 import { Text, View, Button } from 'react-native';
 import { object } from 'prop-types';
+import { isEmpty } from 'lodash';
 
 import LoginForm from 'components/user/LoginForm';
 import { login } from 'actions/userActions';
+import useSessionChangeEffect from 'hooks/useSessionChangeEffect';
 import translate from 'utils/i18n';
-import { SIGN_UP_SCREEN } from 'constants/screens';
+import { SIGN_UP_SCREEN, MAIN_SCREEN } from 'constants/screens';
 import styles from './styles';
 
-const LoginScreen = ({ navigation }) => {
+const LoginScreen = memo(({ navigation }) => {
   const dispatch = useDispatch();
   const loginRequest = useCallback(user => dispatch(login(user)), [dispatch]);
+
+  useSessionChangeEffect(
+    user => {
+      const userLoggedIn = !isEmpty(user);
+      if (userLoggedIn) {
+        navigation.navigate(MAIN_SCREEN);
+      }
+    },
+    [navigation]
```

Comment:
I'm trying to ### understand this, why are you triggering the hook when the navigation object changes?

Reply:
I'm not getting this too 🤔

Reply:
If I remember correctly, it shouldn't, I added it just as a good practice (since all props used inside an effect should be  passed as dependencies). But we can definitely omit it.